server/managers/AudioMetadataManager.js

```diff
@@ -85,6 +89,32 @@ class AudioMetadataMangaer {
       */
 
       const ffmpegOptions = ['-c copy', '-map_chapters 1', '-map_metadata 1', `-metadata track=${audioFile.index}`, '-write_id3v2 1', '-movflags use_metadata_tags']
+
+      if (coverPath != '') {
+        var ffmpegCoverPathInput = {
+          input: coverPath,
+          options: ['-f image2pipe']
+        }
+        var ffmpegCoverPathOptions = [
+          '-c:v copy',
+          '-map 2:v',
+          '-map 0:a'
+        ]
+
+        ffmpegInputs.push(ffmpegCoverPathInput)
+        Logger.debug(`[AudioFileMetaDataManager] Cover found for "${audioFile.metadata.filename}". Cover will be merged to metadata`)
+      } else {
+        // remove the video stream to account for the user getting rid an existing cover in abs
```

Comment:
This isn't necessary actually since this is encoding the metadata into a new audio file which then gets moved and overwrites the existing audio file. See https://github.com/advplyr/audiobookshelf/blob/master/server/managers/AudioMetadataManager.js#L107

The reason for this is because ffmpeg doesn't allow the input and output files to be the same, so we need to write to a new file.

Reply:
Disregard, I see what you mean now. If the audio file already has a cover then if it isn't explicitly removed it will remain with the audio file.